src/test_array_builder.cc

```diff
@@ -944,6 +945,42 @@ bool build_heterogeneous_arrays(
   return true;
 }
 
+void put_metadata(Group& g, const std::string& key, const std::string& value)
+{
+    g.put_metadata(key, TILEDB_STRING_ASCII, static_cast<uint32_t>(value.size()), value.data());
+}
+
+template <class T>
+void put_metadata(Group &g, const std::string &key,
+                  tiledb_datatype_t datatype, T value) {
+  g.put_metadata(key, datatype, 1, &value);
+
+  std::array<T, 7> values;
+  std::fill(values.begin(), values.end(), value);
+  g.put_metadata(key + "_multi", datatype, static_cast<uint32_t>(values.size()), values.data());
+};
+
+void build_group(Context &ctx, const std::string &group_base) {
+  Group g{ctx, group_base, tiledb_query_type_t::TILEDB_WRITE};
+
+  put_metadata<bool>(g, "bool", TILEDB_BOOL, true);
+  put_metadata<uint8_t>(g, "u8", TILEDB_UINT8, 77);
+  put_metadata<uint16_t>(g, "u16", TILEDB_UINT16, 777);
+  put_metadata<uint32_t>(g, "u32", TILEDB_UINT32, 77777);
+  put_metadata<uint64_t>(g, "u64", TILEDB_UINT64, 7777777777);
+  put_metadata(g, "str", "77777");
```

Comment:
Consider making this some number of '7's different from number of digits in any of above numeric values...

Reply:
Done, I used hex values for the integers.